src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -128,11 +120,67 @@ private static AbstractTreeIterator prepareTreeParser(Repository repository, Str
     }
 
     /**
+     * Gets the merge-base of two commits.
```

Comment:
> kept the function name getMergeBase in DiffParser, since it is just doing the related git command git merge-base. 

Please expand this documentation and explain what a merge-base is.